providers/implementations/keymgmt/ec_kmgmt.c

```diff
@@ -148,8 +148,10 @@ int key_to_params(const EC_KEY *eckey, OSSL_PARAM_BLD *tmpl,
 
         if (p != NULL || tmpl != NULL) {
             /* convert pub_point to a octet string according to the SECG standard */
+            point_conversion_form_t format = EC_KEY_get_conv_form(eckey);
+
             if ((pub_key_len = EC_POINT_point2buf(ecg, pub_point,
-                                                  POINT_CONVERSION_COMPRESSED,
+                                                  format,
```

Comment:
It's a little hard to square this change with our promise of a stable libcrypto/provider ABI.
Wouldn't it be "safer" (in that regard) to redefine `OSSL_KEYMGMT_SELECT_PUBLIC_KEY` to always be compressed, and define a new selector flag to get the public key in the form associated with the key (or in uncompressed form specifically)?
In particular, how would a mismatched 3.0.0/3.0.1 libcrypto/default-provider (or vice versa) handle this change?

Reply:
On the other hand, documentation never restricted which SECG format the point would be encoded with, and AFAICT the format for import/export (used to?) be considered a provider specific detail.